airflow/models/dagrun.py

```diff
@@ -829,6 +862,50 @@ def is_effective_leaf(task):
         leaf_tis = {ti for ti in tis if ti.task_id in leaf_task_ids if ti.state != TaskInstanceState.REMOVED}
         return leaf_tis
 
+    @staticmethod
+    def set_dagrun_span_attrs(span: Span | EmptySpan, dag_run: DagRun, dagv: DagVersion):
+        if dag_run._state == DagRunState.FAILED:
```

Comment:
Given this has a dag run object passed in why isn't it a normal method:

```suggestion
    def set_dagrun_span_attrs(self, span: Span | EmptySpan, dagv: DagVersion):
        if self._state == DagRunState.FAILED:
```

Reply:
Changed it to a normal method.